src/sampler.jl

```diff
@@ -1,3 +1,72 @@
+##
+## Interface functions
+##
+
+struct Sample{P<:PhasePoint, NT<:NamedTuple}
```

Comment:
Should this function be named `HMCTransition` according to https://github.com/TuringLang/Turing.jl/blob/csp/interface-pt2/src/interface/Interface.jl#L37

Reply:
I think it a bit and feel `Transition` seems to be a better name. Ideally I don't think we would store the momentum variable in a `Sample` - all we need is position. On the other side `Transition` sounds more wider than `Sample`.

Reply:
Maybe in transition we should return a namedtuple `(z=z, stat...)`? How do you think?

Reply:
I dont’t have A preference here, although I think a Transition type is more consistent with Interface.jl

Reply:
I see. That's not a bit problem I think. I can create a specific type for that named tuple to store transition. I guess this probably the ideal way (as one of reason to use NamedTuple is just for convenience).

Reply:
OK it's not easy either. If we make a new struct rather than using `NamedTuple` we also need to implment some helper functions like `iterate`. Maybe not worth it for now. 

I will just rename it to `Transition` for now.

Reply:
@cpfiffer I updated this PR and push a corresonding update to TuringLang/Turing.jl#793